x-pack/plugins/security_solution/docs/testing/test_plans/detection_response/prebuilt_rules/installation_and_upgrade.md

```diff
@@ -732,6 +734,145 @@ And the Investigation Guide tab should NOT be displayed
 
 TODO: add scenarios https://github.com/elastic/kibana/issues/166215
 
+### Rule upgrade workflow: viewing rule changes before upgrading
+
+#### **Scenario: User can see changes in a side-by-side view**
+
+**Automation**: 1 Cypress e2e test
+
+```Gherkin
+Given X prebuilt rules are installed in Kibana
+And for Y of these rules new versions are available
+When user opens the Rule Updates table and selects a rule
+Then the upgrade preview should open
+And rule changes should be displayed in a two-column view
+And correct rule version numbers should be displayed in their respective columns
+When the user selects another rule without closing the preview
+Then the preview should display the changes for the newly selected rule
+```
+
+#### **Scenario: User can see precisely how property values would change after upgrade**
+
+**Automation**: 1 UI component-level test
+
+```Gherkin
+Given a rule preview with rule changes is open
+Then each line of <column> that was <change_type> should have <bg_color> background
+And marked with <line_badge> badge
+And each changed word in <column> should be highlighted with <accent_color>
+
+Examples:
+| change_type | column       | bg_color         | accent_color         | line_badge |
+| updated     | Base version | removed_bg_color | removed_accent_color | -          |
+| updated     | Update       | added_bg_color   | added_accent_color   | +          |
+| removed     | Base version | removed_bg_color | none                 | -          |
+| removed     | Update       | none             | none                 | none       |
+| added       | Base version | none             | none                 | none       |
+| added       | Update       | added_bg_color   | none                 | +          |
+```
+
+#### **Scenario: Rule actions and exception lists should not be shown as modified**
+
+**Automation**: 1 UI component-level test
+
+```Gherkin
+Given a prebuilt rule is installed in Kibana
+And this rule doesn't have any actions or an exception list
+And a user has set up actions and an exception list for this rule
+And this rule has an update available
+And the update doesn't define any actions or an exception list
+When a user opens the upgrade preview for this rule
+Then the preview should open
+And there should not be any modifications to rule's actions or exception list
+```
+
+#### **Scenario: Stateful properties should not be included in preview**
```

Comment:
My plan was to keep technical and dynamic properties separate, because I thought it would make sense to test dynamic properties using some kind of E2E test - install a rule, execute it once, then see if update lists only the properties we expect. Keep it more real-life since it relates to runtime changes.

And as for technical properties, we could just test them with something like react-testing-library and two harcoded rule objects. What do you think?

Reply:
@nikitaindik sounds good to me 👍